src/Repository/AnnouncementRepository.php

```diff
@@ -18,64 +18,81 @@ class AnnouncementRepository extends ServiceEntityRepository
   public function __construct(ManagerRegistry $registry)
   {
     parent::__construct($registry, Announcement::class);
+
   }
 
   /**
    * Custom method to pull all announcements created for the current date
    * 
    * @author Daniel Boling
    */
-  public function find_today()
+  private function query_today($date_input = 'now', $approval = 1)
   {
 
-    $date = new \DateTime('now', new \DateTimeZone('America/Indiana/Indianapolis'));
+    $date = new \DateTime($date_input, new \DateTimeZone('GMT'));
 
-    return $this->createQueryBuilder('a')
-      ->andWhere('a.start_date <= :date AND a.end_date >= :date')
-      ->setParameter('date', $date->format('Y-m-d'))
+    $qb = $this->createQueryBuilder('a')
+      ->Where('a.start_date <= :date AND a.end_date >= :date');
+    if ($approval == 1)
+    {
+      $qb->andWhere('a.approval = 1');
+    }
+    $qb->setParameter('date', $date->format('Y-m-d'))
       ->orderBy('a.id', 'ASC')
-      ->getQuery()
+    ;
+    return $qb->getQuery();
+
+  }
+
+  /**
+   * Method using query_today to get a result array.
+   * Requirements: date match.
+   * 
+   * @author Daniel Boling
+   */
+  public function find_today($date_input = 'now', $approval = 1)
+  {
+
+    return $this->query_today($date_input, $approval)
       ->getResult()
     ;
 
   }
 
     /**
-   * Custom method to pull announcements created for the given date
+   * Method using query_today to get a countable array.
    * 
    * @author Daniel Boling
    */
-  public function find_by_day($date)
+  public function count_today($date_input = 'now', $approval = 1)
   {
 
-    $date = new \DateTime($date, new \DateTimeZone('America/Indiana/Indianapolis'));
-
-    return $this->createQueryBuilder('a')
-      ->andWhere('a.start_date <= :date AND a.end_date >= :date')
-      ->setParameter('date', $date->format('Y-m-d'))
-      ->orderBy('a.id', 'ASC')
-      ->getQuery()
+    return $this->query_today($date_input, $approval)
       ->getScalarResult()
     ;
 
   }
 
   /**
-   * Custom method purely for testing to return the last-inserted result.
+   * Method using query_today to get a result array.
+   * Requirement: date match. Approved.
    * 
    * @author Daniel Boling
    */
-  public function findLastInserted()
+  public function find_today_approved($date_input = 'now')
   {
 
+    $date = new \DateTime($date_input, new \DateTimeZone('GMT'));
+
     return $this->createQueryBuilder('a')
-      ->orderBy('a.id', 'DESC')
-      ->setMaxResults(1)
+      ->where('a.start_date <= :date AND a.end_date >= :date')
+      ->andWhere('a.approval = 1')
+      ->setParameter('date', $date->format('Y-m-d'))
+      ->orderBy('a.id', 'ASC')
       ->getQuery()
-      ->getOneOrNullResult()
+      ->getResult()
     ;
   }
```

Comment:
I don't think this function is needed anymore since `find_today()` can handle both approved and unapproved announcements, right?